### ISSUE TITLE
Add times to Activities -- either duration or start/end

### ISSUE THREAD

**Issue by mgrhm** [monicahq/monica#5221]
I'd really like to be able to (optionally) add a start and end time to Activities.  It would be useful for making Monica a more complete record.  It would also avoid the strange situation where Monica will occasionally mis-sort Activities on the same day -- sometimes putting the earlier activity before the later one.

**Comment by weaversam8** [comment]
Great idea! I think it's tough to balance precision with ease of entry though - a feature like this should include some vague times (morning / lunch / evening) in addition to allowing someone to input a precise time.

**Comment by mgrhm** [comment]
Agreed.  It is difficult to balance precision with ease of entry.  I like the idea of being able to tag an Activity with a more broad 'time of day'.  It's similar-to but distinct-from the Activity's category.

I'm not sure how you've implement these, though.  Two separate options when adding the Activity?  "Add a time of day" and "Add a start and end time"?  That's reasonable clear (wording could maybe do with some work), but obviously risks adding more clutter.

**Comment by weaversam8** [comment]
I'd implement an option called "Add a time" to the activity, which would pop up a "time of day" dropdown first, with a button (or dropdown option) to be more specific.

![wireframe depicting an "add a time" button, with an arrow showing that button pointing towards a "time of day" dropdown, with "morning", "afternoon", "evening", "night", and "custom" options. The "custom" option has an arrow that points towards a pair of textboxes which read "10:30 AM to 1:00 PM"](https://user-images.githubusercontent.com/2546219/120331552-e3b69a00-c2bb-11eb-8997-4c3af07a690a.png)

**Comment by mgrhm** [comment]
By preference, I think I'd split it into two distinct options to reduce the amount clicking I'd have to do to enter a start/end time.  That's based on my own preference and my own preference for exact times over fuzzy times.

Extending this idea a little further, eventually Monica could provide a calendar feed with these events populated.

**Comment by weaversam8** [comment]
That's fair. I still think that exact times are a bit overkill for the average Monica user, but it wouldn't be hard to add a user preference / setting in Monica that defaulted to the exact time interface when you clicked "Add a time" with a button to go back to fuzzy times if necessary.

**Comment by mgrhm** [comment]
It's definitely overkill for the average user, but some people probably wouldn't use fuzzy times.  (I probably wouldn't.)

Being able to switch back and forth like you describe seems like a good solution.
